haystack/preview/components/generators/openai/gpt.py

```diff
@@ -121,7 +121,7 @@ def from_dict(cls, data: Dict[str, Any]) -> "GPTGenerator":
         """
         init_params = data.get("init_parameters", {})
         streaming_callback = None
-        if "streaming_callback" in init_params:
+        if "streaming_callback" in init_params and init_params["streaming_callback"]:
```

Comment:
This bug was found when I introduced serialization for the RAG pipelines. Maybe worth adding a unit test too?

Reply:
Yes please, in a separate PR.